### PR TITLE
Allow to extend Apollo Client and network interface config, more tests (client config, query deduplication)

### PR DESCRIPTION
**Pull request by xavcz**
It's me again! 😄  

[Query deduplication](http://dev.apollodata.com/core/network.html#query-deduplication) can be pretty handy, I've thought it would be a good thing to offer Meteor developers using this package to set deduplication option up.

I believe this open the door to a new way to configure the Apollo Client via the Meteor `apollo` package, [given all the params an Apollo Client instance accepts](https://github.com/apollographql/apollo-client/blob/master/src/ApolloClient.ts#L104-L134). 

I see especially two types of config:
* Network interface config
* And the rest 😄 (we are already using `ssrMode`, `queryDeduplication` & `dataObjectFromId`)

We could have two types of default config, the `defaultNetworkInterfaceConfig` and `defaultClientOptions`? Or maybe everything in the same object? 

I don't know what would be the easiest and the most useful for Meteor folks. What do you think?

### CODE REVIEW

File: main-client.js [hunk 1]
```diff
@@ -80,8 +80,12 @@ export const createMeteorNetworkInterface = (givenConfig) => {
   return networkInterface;
 };
 
-export const meteorClientConfig = (networkInterfaceConfig) => {
+export const meteorClientConfig = (globalConfig = {}) => {
+  // destructure general apollo client configuration from network interface configuration
+  const { queryDeduplication = false, ...networkInterfaceConfig } = globalConfig;
```
Comment (lorensr) [inline main-client.js]: isn't networkInterfaceConfig now undefined?

**Comment by lorensr** [comment]
Two objects I think.

So should we change the API from:

`meteorClientConfig(networkInterfaceConfig)`

to:

`meteorClientConfig(clientConfig)`

and `clientConfig` can take any Apollo Client options plus `networkInterfaceConfig`? Which would be overwritten if `clientConfig.networkInterface` were included? 

And we can explain what defaults we use

File: main-client.js (hunk 1 above)
Comment (xavcz) [inline main-client.js]: I got scared at first! 😱 It's has intended not `undefined` but maybe it should be done some other way to be clearer ?

Something approximatively like: 
```js
export const meteorClientConfig = ({
  networkInterface, // expected config object by Apollo Client
  ...clientConfig,  
} = {
  networkInterface: createMeteorNetworkInterface()
}) => {

...

}
```

**Comment by xavcz** [comment]
Yep totally!

File: main-client.js (hunk 1 above)
Comment (lorensr) [inline main-client.js]: oh haha sorry i wasn't paying attention

**Comment by lorensr** [comment]
Want to do that to this PR & docs changes? Or separately?

**Comment by xavcz** [comment]
I'll do the modifications on this one 👍

**Comment by xavcz** [comment]
yay

💃

**Pull request merged**